src/Cassandra/Connections/Connection.cs

```diff
@@ -816,6 +817,8 @@ private void RunWriteQueueAction()
                     RemoveFromPending(streamId);
                     //Callback with the Exception
                     state.InvokeCallback(ex);
+
+                    stream?.SetLength(startPosition);
                     break;
```

Comment:
@jorgebay can you think of any reason for this `break` being here? Shouldn't the driver keep writing "valid" requests to the stream and skip the "invalid" ones?

Reply:
hm... good question...
It avoids further coalescing of the message and the valid messages will continue to be written (`socket.Write()` and `RunWriteQueue()` calls below), but I understand that it's not needed.

It doesn't have any real impact (outside having smaller messages when there is a error building an individual request) so we can evaluate in future revisions whether we want to take it out.

Reply:
done

Reply:
Just make sure if this `break` is removed that it doesn't fall through to:

```
state.Request = null;
totalLength += frameLength;
```

we don't want to incr `totalLength`

Reply:
yes, I don't think its a good idea to change the existent logic in the context of this bug fix.